src/extensionsIntegrated/TabBar/main.js

```diff
@@ -128,23 +122,42 @@ define(function (require, exports, module) {
         const activePathInPane = activeFileInPane ? activeFileInPane.fullPath : null;
 
         // Check if this file is active in its pane
-        const isActive = (entry.path === activePathInPane);
+        const isActive = entry.path === activePathInPane;
 
         // Current active pane (used to determine whether to add the blue underline)
         const currentActivePane = MainViewManager.getActivePaneId();
-        const isPaneActive = (paneId === currentActivePane);
+        const isPaneActive = paneId === currentActivePane;
 
         const isDirty = Helper._isFileModified(FileSystem.getFileForPath(entry.path));
         const isPlaceholder = entry.isPlaceholder === true;
 
+        let gitStatus = ""; // this will be shown in the tooltip when a tab is hovered
+        let gitStatusClass = ""; // for styling
+
+        if (window.phoenixGitEvents && window.phoenixGitEvents.TabBarIntegration) {
+            const TabBarIntegration = window.phoenixGitEvents.TabBarIntegration;
+
+            // find the Git status
+            // if untracked we add the git-new class and U char
+            // if modified we add the git-modified class and M char
+            if (TabBarIntegration.isUntracked(entry.path)) {
+                gitStatus = "Untracked";
+                gitStatusClass = "git-new";
+            } else if (TabBarIntegration.isModified(entry.path)) {
+                gitStatus = "Modified";
+                gitStatusClass = "git-modified";
+            }
+        }
+
         // create tab with all the appropriate classes
         const $tab = $(
             `<div class="tab 
-            ${isActive ? 'active' : ''} 
-            ${isDirty ? 'dirty' : ''}
-            ${isPlaceholder ? 'placeholder' : ''}" 
+            ${isActive ? "active" : ""}
```

Comment:
Avoid using full file beautify as part of pull requests as it create too many non related changes to review that is hard to review.